src/components/TaskSupplier.js

```diff
@@ -0,0 +1,123 @@
+import {useState} from 'react';
+import AddTask from './AddTask';
+import Tasks from './Tasks';
+import Footer from './Footer';
+import './TaskSupplier.css';
+
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import {useCollectionData} from "react-firebase-hooks/firestore";
+import { collection, deleteDoc, doc, query, serverTimestamp, setDoc, orderBy, updateDoc } from "firebase/firestore";
+
+const collectionName = "List-Collection";
+const subCollectionName = "Tasks-Collection";
+
+export default function TaskSupplier(props) {
+
+    const [showComplete, setShowComplete] = useState(false);
+    const [sortBy, setSortBy] = useState("created");
+
+    const qTask = query(collection(props.db, collectionName, props.currentListId, subCollectionName), orderBy(sortBy))
+    const [tasks, loadingTasks, errorTasks] = useCollectionData(qTask);
+
+    // Add task
+    function addTask (taskName) {
+        const uniqueId = generateUniqueID();
+        setDoc(doc(props.db, collectionName, props.currentListId, subCollectionName, uniqueId),
+            {
+                id: uniqueId,
+                text: taskName,
+                complete: false,
+                priorityLevel: 1,
+                created: serverTimestamp(),
+            });
+    }
+
+    // Delete task
+    function deleteCompletedTasks () {
+
+        tasks.forEach(task => task.complete && deleteDoc(doc(props.db, collectionName, props.currentListId, subCollectionName, task.id)));
+    }
+
+    // Hide Task
+    function hideTask () {
+        setShowComplete(!showComplete)
+    }
+
+    // Rename Task
+    function renameTask (id, value) {
+        updateDoc(doc(props.db, collectionName, props.currentListId, subCollectionName, id), {text: value});
+    }
+
+    // Complete Task
+    function completedTask (id, value) {
```

Comment:
Better name: completeTask (parellel with renameTask)

Reply:
Very odd that the incoming value is notted before use.  I'd suggest instead:
completeTask(taskId, isCompleted) where isCompleted is the new desired value for completed.